src/marqo/core/embed/embed.py

```diff
@@ -0,0 +1,123 @@
+from marqo.core.models.marqo_index import MarqoIndex, FieldType, UnstructuredMarqoIndex, StructuredMarqoIndex
+from marqo.core.models.marqo_query import MarqoTensorQuery, MarqoLexicalQuery
+from marqo.core.structured_vespa_index.structured_vespa_index import StructuredVespaIndex
+from marqo.core.unstructured_vespa_index import unstructured_validation as unstructured_index_add_doc_validation
+from marqo.core.unstructured_vespa_index.unstructured_vespa_index import UnstructuredVespaIndex
+from marqo.core.vespa_index import for_marqo_index as vespa_index_factory
+from marqo.s2_inference import errors as s2_inference_errors
+from marqo.s2_inference import s2_inference
+from marqo.s2_inference.clip_utils import _is_image
+from marqo.s2_inference.processing import image as image_processor
+from marqo.s2_inference.processing import text as text_processor
+from marqo.s2_inference.reranking import rerank
+from marqo.tensor_search import delete_docs
+from marqo.tensor_search import enums
+from marqo.tensor_search import index_meta_cache
+from marqo.tensor_search import utils, validation, add_docs, tensor_search
+from marqo.tensor_search.enums import (
+    Device, TensorField, SearchMethod, EnvVars
+)
+from marqo.tensor_search.index_meta_cache import get_cache, get_index
+from marqo.tensor_search.models.add_docs_objects import AddDocsParams
+from marqo.tensor_search.models.api_models import BulkSearchQueryEntity, ScoreModifier
+from marqo.tensor_search.models.delete_docs_objects import MqDeleteDocsRequest
+from marqo.tensor_search.models.private_models import ModelAuth
+from marqo.tensor_search.models.search import Qidx, JHash, SearchContext, VectorisedJobs, VectorisedJobPointer
+from marqo.tensor_search.telemetry import RequestMetricsStore
+from marqo.tensor_search.tensor_search_logging import get_logger
+from marqo.vespa.exceptions import VespaStatusError
+from marqo.vespa.models import VespaDocument, FeedBatchResponse, QueryResult
+
+logger = get_logger(__name__)
+
+
+class Embed:
+    def __init__(self, config):
+        self.config = config
+
+    def embed_content(
+                    self, content: Union[Union[str, Dict[str, float]], List[Union[str, Dict[str, float]]]],
+                    index_name: str, device: str = None,
+                    image_download_headers: Optional[Dict] = None,
+                    model_auth: Optional[ModelAuth] = None
+                    ) -> List[List[float]]:
+        """
+        Use the index's model to embed the content
+
+        Returns:
+                List of embeddings corresponding to the content. If content is a list, the return list will be in the same order.
+                If content is a string, the return list will only have 1 item.
+        """
+
+        # Content validation is done in API model layer
+        t0 = timer()
+
+        # Determine device
+        if device is None:
+            selected_device = utils.read_env_vars_and_defaults("MARQO_BEST_AVAILABLE_DEVICE")
```

Comment:
Can you pass this in the constructor? We don't want to access env vars outside of the API. It reduces your code's reusability and it's also less explicit (since you're reading configuration outside of class initialization and method signature)

Reply:
done